imagebuildah/build.go

```diff
@@ -1559,6 +1559,10 @@ func processCopyFrom(dockerfiles []io.ReadCloser) []io.ReadCloser {
 	// asMap contains the names of the images seen after a "FROM image AS"
 	// line in the Dockefiles.  The boolean value just completes the map object.
 	asMap := make(map[string]bool)
+	// foreignImages is the list of images that are referenced as sources
+	// of "COPY --from" instructions, but which are not the result of
+	// earlier stages in the build.
+	foreignImages := []string{}
```

Comment:
I think it's cheaper and faster to make `foreignImages` a `map[string]bool` and iterate over the keys down in line 1626.

Reply:
A version of this patch did that, but I went with this version to make sure that we pull the images in same order in which they are first referenced in the Dockerfile.

Reply:
Makes sense, thanks!